packages/bitbucket-eslint-report/index.ts

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env node
+
+import fs from 'fs/promises';
+import path from 'path';
+import commandLineArgs from 'command-line-args';
+import {
+  uploadReportToBitbucket,
+  uploadAnnotationsToBitbucket,
+  createBuildOnBitbucket,
+} from '@zooshdigital/bitbucket-code-insights';
+import { BitbucketAnnotation, BitbucketSeverity } from '@zooshdigital/bitbucket-code-insights/dist/types';
+
+import { createLogger } from 'utils/logger';
+
+type LintMessage = {
+  ruleId: string;
+  severity: number;
+  line: number;
+  column: number;
+  fix?: {
+    text: string;
+  };
+};
+
+type LintFile = {
+  filePath: string;
+  errorCount: number;
+  fixableErrorCount: number;
+  warningCount: number;
+  fixableWarningCount: number;
+  messages: LintMessage[];
+};
+
+const optionDefinitions = [
+  { name: 'name', alias: 'n', type: String },
+  { name: 'path', alias: 'p', type: String, multiple: true },
+  { name: 'add-build', alias: 'a', type: Boolean },
+  { name: 'strict', alias: 's', type: Boolean },
+];
+
+const args = commandLineArgs(optionDefinitions);
+
+const logger = createLogger('Zoosh Bitbucket Eslint Report');
+
+const severities: Record<number, BitbucketSeverity> = {
+  0: 'LOW',
+  1: 'MEDIUM',
+  2: 'HIGH',
+};
+
+async function uploadReport() {
+  const items: BitbucketAnnotation[] = [];
+
+  let counter = 0;
+  let totalErrors = 0;
+  let fixableErrors = 0;
+  let totalWarnings = 0;
+  let fixableWarnings = 0;
+
+  try {
+    const { name, path: reportPaths, 'add-build': addBuild, strict } = args;
+
+    if (!name || !reportPaths || reportPaths.length === 0) {
+      throw new Error(
+        'Bitbucket Eslint Report - Usage: bitbucket-eslint-report -n <report-name> -p <report-path> [-b] [-s]',
+      );
+    }
+
+    await Promise.all(
+      reportPaths.map(async (reportPath: string) => {
+        const lint: LintFile[] = JSON.parse(await fs.readFile(reportPath, 'utf8'));
+        lint.forEach((file) => {
+          // const filePath = file.filePath.replace(/^\/opt\/atlassian\/pipelines\/agent\/build\//, '');
```

Comment:
Unused code can be removed

Reply:
In this case I'd keep this as an exception. In any case that the actual process.cwd replacement wouldn't work, we could still fall back to the hard path replacement, as that should be static in the bitbucket pipelines, and it's good to remember what that is. Or I can put at least a comment above.

Reply:
Then, please add a comment to this part to explain why it is there.

Reply:
I did already! :D

Reply:
okay, thank you!